### PR TITLE
docs: add docs site built with Astro

### PR DESCRIPTION
**Pull request by kroucher**
# Add Documentation site built with Astro

- [x] I reviewed linter warnings + errors, resolved formatting, types and other issues related to my work
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

This pull request is to include a documentation site in the create-t3-app repo.

There are instructions in `/docs/CONTRIBUTING.md` that explains where files are located to be modified.

Built using:

- Astro
- Tailwind (Responsive ✅, Dark Mode ✅)
- React
- TypeScript

Demo deployed at https://create-t3-app-docs.vercel.app/

---

## Screenshots

image

💯

### CODE REVIEW

**Comment by nexxeln** [comment]
I think you should make this PR in the [create.t3.gg](https://github.com/t3-oss/create.t3.gg) repo

**Comment by kroucher** [comment]
Can do! 👍

**Comment by juliusmarminge** [comment]
> I think you should make this PR in the [create.t3.gg](https://github.com/t3-oss/create.t3.gg) repo

Would it not be better to keep everything in one place though?

**Comment by juliusmarminge** [comment]
Like I said in https://github.com/kroucher/create-t3-app-docs/issues/1, it would boost visibility and probably lead to more people contributing to the docs if it would be in this repo. @nexxeln

**Comment by juliusmarminge** [comment]
We can kill three birds in one stone here: #210, #231 and docs. What'ya think @nexxeln

**Comment by nexxeln** [comment]
Agree, let's work on it

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+yarn-error.log*
+pnpm-debug.log*
+
+
+# environment variables
+.env
+.env.production
+
+# macOS-specific files
+.DS_Store
```
Comment (juliusmarminge) [inline docs/.gitignore:19]: should be good with the root .gitignore

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# [create-t3-app](https://github.com/t3-oss/create-t3-app): Docs Site
```
Comment (juliusmarminge) [inline docs/README.md:1]: Lets rename it `www` since we will have more stuff than just docs on here

File: docs/astro.config.mjs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export default defineConfig({
+  integrations: [
+    react(),
+    tailwind({
+      config: {
+        applyBaseStyles: false,
+      },
+    }),
+  ],
+  site: `https://create-t3-app-docs.vercel.app/`,
```
Comment (juliusmarminge) [inline docs/astro.config.mjs:15]:
```suggestion
  site: "https://create.t3.gg",
```
we should use this domain, right @nexxeln @TheoBr

File: docs/lib/updateAlgolia.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
+"use strict";
```
Comment (juliusmarminge) [inline docs/lib/updateAlgolia.js:1]: is it necessary with both a `.js` and a `.ts` version here?

**Comment by juliusmarminge** [comment]
Merging this into a branch on the repo and we can iterate from there. Giving you perms too @kroucher

**Pull request merged**